cargo-near/src/build.rs

```diff
@@ -51,6 +45,12 @@ pub(crate) fn run(args: BuildCommand) -> anyhow::Result<()> {
             )?;
             min_abi_path.replace(util::copy(&path, &out_dir)?);
         }
+        contract_abi.metadata.build = Some(BuildInfo {
+            compiler: format!("rustc {}", rustc_version::version()?),
+            builder: format!("cargo-near {}", env!("CARGO_PKG_VERSION")),
+            image: None,
```

Comment:
> Env var only set when built from contract-builder?

Most likely. There no built-in way to get the docker image name from inside of the container from what I have researched.

Reply:
I wonder if `sdk-js` gets an equivalent of `contract-builder` at some point and what that would mean for this field. I guess in a sense, the `builder` information plus the `image` helps better describe what type of contract-builder this is. But it's a non-issue, we have a long way ahead before we're at that point.

Reply:
I was actually wondering if it makes sense to "demote" `contract-builder` to only serve as an operating system and then just install whatever is necessary based on the build info (e.g. rustc 1.61.0, cargo-near 0.2.0). This way it could potentially work for both sdk-rs and sdk-js.

Also, presumably sometime in the future rustc/cargo will reach the point where wasm builds can be bit-by-bit reproducible and hedonistic and we will not need an image anymore.

Reply:
This is a good point, not sure what the size overhead would be from including the JS SDK in the same docker image, but it would certainly help unify things. Also, imho, the current docker image might be a little too big, I'll try to see if we can cut things down a bit, drop unnecessary binaries, strip some maybe etc. Doesn't seem like the official rust image makes much of an effort to reduce size anyway.